generator/internal/api/model.go

```diff
@@ -226,6 +297,64 @@ type Message struct {
 	IsPageableResponse bool
 	// PageableItem is the field to be paginated over.
 	PageableItem *Field
+
+	// The API that this message belongs to.
+	API *API
+
+	// Elements is a map of all the elements in the message, keyed by their name.
+	// This field is only available after the message has been visited by the CrossReferencingVisitor.
+	Elements map[string]*MessageElement
```

Comment:
Consider  `ElementsByName map[string]*MessageElement`.  That seems more consistent with the existing `API.State.*ByID` maps.